### ISSUE TITLE
 How remove Markers from the map?

### ISSUE THREAD

**Issue by zs-dima** [gauris26/flutter_animarker#15]
Thanks for the nice library.
For the clustering and filtering markers could be useful to remove unneeded markers from the map on rebuild widget with new markers set.
It works for `GoogleMap(markers:{})` itself but do not work for the `Animarker(markers:{})`

**Comment by X-SLAYER** [comment]
did you find any solutions for this !

**Comment by WagnerMagnus1** [comment]
Nothing yet, I also encountered the same problem.

**Comment by narendra-vaniya** [comment]
Same Here. when I trying to remove marker from `Animarker(markers:{})` it's not working and showing all markers but when I use `GoogleMap(markers:{})` it works properly.
But I found one solution for **hiding markers** on the map. I use `visible:false` property in `Marker()` to hide some marker on map and it work for me.

**Comment by fateh491989** [comment]
I encountered the same problem

**Comment by gauris26** [comment]
I post a new beta update with this bug fixed. Please try it, and let know if working correctly.

**Comment by marcusvpais** [comment]
I managed to remove the marker by using visible: false on it.

**Comment by zs-dima** [comment]
> I managed to remove the marker by using visible: false on it.

It is not good workaround it does not fit most cases

**Comment by ivangr1** [comment]
> I post a new beta update with this bug fixed. Please try it, and let know if working correctly.

I had the same issue and tried your new update but it doesn't help much.
I have many markers moving on a map and need an option to filter them.

This update only works if I want to remove all markers with `markers.clear()`
I tried clearing the markers and then showing them filtered but after they are cleared I cannot make them appear again.

**Comment by zs-dima** [comment]
@gauris26 thanks a lot for looking into
I could not see Ripple effect after the new beta update

![image](https://user-images.githubusercontent.com/25195453/124971925-71f80b80-e032-11eb-9cb9-49d1abd90def.png)

I tested on Android emulator

Flutter 2.4.0-0.0.pre • channel dev
Framework • revision 96bbcd006f (8 days ago) • 2021-06-30 03:43:41 -0400
Engine • revision c633b2a6a0
Tools • Dart 2.14.0 (build 2.14.0-258.0.dev)

  google_maps_flutter: ^2.0.6
  flutter_animarker: ^3.3.1-beta.4

**Comment by manpreetAulakh** [comment]
@ivangr1  solution worked for me , could you please merge his solution into master branch. It will be very helpful for everyone, I think.

**Comment by gauris26** [comment]
If it worked, please PR it.

**Comment by gauris26** [comment]
> > I post a new beta update with this bug fixed. Please try it, and let know if working correctly.
> 
> I had the same issue and tried your new update but it doesn't help much.
> Nothing happens when I add or remove markers from the set.
> It only works if I remove all markers with `markers.clear()`.
> 
> I have many markers moving on a map and need an option to filter them.
> I tried clearing the markers and then showing them filtered but after they are cleared I cannot make them appear again.
> 
> EDIT:
> I managed to solve my problem by changing this code:
> https://github.com/gauris26/flutter_animarker/blob/171c708ca71f2b2331fb5e4b7b71ee8ed5c15dd7/lib/widgets/animarker.dart#L365-L368
> 
> To this:
> 
> ```
> if (widget.markers.length < oldWidget.markers.length) {
>         widget.updateMarkers(oldWidget.markers, widget.markers);
>         return;
>       }
>       if (widget.markers.length > oldWidget.markers.length) {
>         _markers.clear();
>       }
> ```

If it worked, please PR it.

**Comment by zs-dima** [comment]
@ivangr1 @gauris26 
`if (widget.markers.length < oldWidget.markers.length)`
 - different length check will not help
it could be the same length but one of markers could be changed

**Comment by ivangr1** [comment]
> @ivangr1 @gauris26
> `if (widget.markers.length < oldWidget.markers.length)`
> ! different length check will not help
> it could be the same length but one of markers could be changed

Yes, you are right. I tried every other combination and this is the only thing that solves at least one problem. I cannot manage to find a complete solution. Can you help us @gauris26?

**Comment by DSMalaviya** [comment]
How to remove previous markers from map??

**Comment by gauris26** [comment]
Hi guys,

Please try the latest prerelease beta version I submitted in pub.dev, waiting for your feedback.

**Comment by justsaagar** [comment]
> Hi guys,
> 
> Please try the latest prerelease beta version I submitted in pub.dev, waiting for your feedback.

`flutter_animarker: ^3.4.0-beta.1` this one?

**Comment by Hari-iLapz** [comment]
In `flutter_animarker: ^3.2.0` when i update the markers like remove one marker and add two markers it is not reflecting,

in `flutter_animarker: ^3.4.0-beta.1` this version when i update the marker all markers are removed but newly added markers doesn't shown up 

Anybody have solution?

**Comment by Hackoak** [comment]
In flutter_animarker: ^3.2.0 when i update the markers like remove one marker and add two markers it is not reflecting, newly added markers are showing but deleted markers also showing

in flutter_animarker: ^3.4.0-beta.1 this version when i update the marker all markers are removed but newly added markers doesn't shown up

Anybody have solution?

**Comment by yncodes** [comment]
> ould not see Ripple effect after the new beta up

same

**Comment by binemmanuel** [comment]
> > I post a new beta update with this bug fixed. Please try it, and let know if working correctly.
> 
> I had the same issue and tried your new update but it doesn't help much. Nothing happens when I add or remove markers from the set. It only works if I remove all markers with `markers.clear()`.
> 
> I have many markers moving on a map and need an option to filter them. I tried clearing the markers and then showing them filtered but after they are cleared I cannot make them appear again.
> 
> EDIT: I managed to solve my problem by changing this code:
> 
> https://github.com/gauris26/flutter_animarker/blob/171c708ca71f2b2331fb5e4b7b71ee8ed5c15dd7/lib/widgets/animarker.dart#L365-L368
> 
> To this:
> 
> ```
> if (widget.markers.length < oldWidget.markers.length) {
>         widget.updateMarkers(oldWidget.markers, widget.markers);
>         return;
>       }
>       if (widget.markers.length > oldWidget.markers.length) {
>         _markers.clear();
>       }
> ```

I tried this package out today and encountered this issue. if I change the visibility or remove a marker the marker remains on the map regardless.

I want to believe that it is safe to do this to fix the issue;
```dart
if (widget.markers != oldWidget.markers) {
      widget.updateMarkers(oldWidget.markers, widget.markers);
      return;
}
```

### EDIT
After trying it out, it works just fine for me
